.buildkite/pipeline_cross.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Generate Buildkite Cross Snapshot/Restore pipelines dynamically
+
+1. Generate snapshots for each instance and kernel version
+2. wait
+3. Restore snapshots across instances / kernels
+"""
+
+import itertools
+
+from common import DEFAULT_PLATFORMS, group, pipeline_to_json
+
+
+def restore_step(label, src_instance, src_kv, dst_instance, dst_os, dst_kv):
+    """Generate a restore step"""
+    return {
+        "command": [
+            f"buildkite-agent artifact download snapshots/{src_instance}_{src_kv}/* .",
+            f"mv -v snapshots/{src_instance}_{src_kv} snapshot_artifacts",
+            "./tools/devtool -y test -- -ra -m nonci integration_tests/functional/test_snapshot_restore_cross_kernel.py",
+        ],
+        "label": label,
+        "timeout": 30,
+        "agents": [f"instance={dst_instance}", f"kv={dst_kv}", f"os={dst_os}"],
+    }
+
+
+def cross_steps():
+    """Generate group steps"""
+    snap_instances = ["m5d.metal", "m6i.metal", "m6a.metal"]
+    groups = []
+    commands = [
+        "./tools/devtool -y create_snapshot_artifacts",
+        "mkdir -pv snapshots/{instance}_{kv}",
+        "sudo chown -Rc $USER: snapshot_artifacts",
+        "mv -v snapshot_artifacts/* snapshots/{instance}_{kv}",
+    ]
+    groups.append(
+        group(
+            "📸 create snapshots",
+            commands,
+            timeout=30,
+            artifact_paths="snapshots/**/*",
+            instances=snap_instances,
+            platforms=DEFAULT_PLATFORMS,
+        )
+    )
+    groups.append("wait")
+
+    kvs = ["linux_4.14", "linux_5.10"]
+    instances_x86_64 = ["m5d.metal", "m6i.metal", "m6a.metal"]
```

Comment:
Is this going to test AMD <-> Intel snapshots too? I don't think it is supported?

Reply:
Yeah, m6i -> m6a doesn't work... though m6a -> m6i seems to work. Should we skip the tests completely or assert that they fail/work when appropiate?

Reply:
I don't think it makes much sense to assert they fail. I guess we can have an exception list of what we know is not working (either due to limitations or by design).

Reply:
done